### PR TITLE
Post Settings Design Improvements & Refactors

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR makes several improvements to the Post Settings both UI and code-wise. Here is a list of major changes:

1. The design of the page is finally updated, aside from Location setting, we are pretty close to it's final form. Location changes will be done on a separate PR.
2. AlertDialogs are changed to use the Calypso style and AppCompat version is used now to support the lower apis properly
3. String resources re-organized and refactored. I opted for more verbose naming here. I thought that'd be better for the translators and would be easier to locate for devs.

To test:
* Since this is a design improvement & refactor PR, there is nothing specific to test aside from checking the Post Settings page out as a whole

Please note that this is not the final form. There probably will be a polish PR coming where we work with a designer for the final touches.

P.S: Even though the PR looks big, all changes are pretty straightforward. I didn't want to break this up too much especially since we have a time-zone difference with @tonyr59h.

screenshot_1497367150

### CODE REVIEW

**Comment by oguzkocer** [comment]
After our discussions yesterday with @mattmiklic, we have decided to go with a card-view layout. With these changes we're pretty close to wrapping up the design changes to the existing features. Here are a few screenshots from the latest version:

screenshot_1497444016
screenshot_1497444024
screenshot_1497444079

As mentioned on the original PR description, I'll handle the location changes in the following PR. @mattmiklic Could you take a look at the screenshots and let me know if it looks OK? I think we might want to change the "Set Featured Image" button a bit. It had white background in the original design, but with this new white background layout we have, it looked horrible, so I didn't do that.

@tonyr59h This should be ready to review, it'd be awesome if you can get to it today or let me know so I can ping someone else as it's a big blocker for the rest of the changes.

**Comment by mattmiklic** [comment]
@oguzkocer That's looking great to me! The only thing I see that's off is the Location setting -- if you look at the mockup on #5811, it looks like the Tags item -- if the location isn't set, it just says "Not set", and if it is set, the location should be displayed.

**Comment by oguzkocer** [comment]
> As mentioned on the original PR description, I'll handle the location changes in the following PR. 

@mattmiklic ^^. That's incoming :)

**Comment by tonyr59h** [comment]
This screen is looking better every day, I love the card-based design! If everything is functioning and there are no new issues it sounds like we should merge it as-is and any design-related issues will be handled in future PRs.

**Comment by tonyr59h** [comment]
My API 16 emulator crashes when loading the settings screen. The logs aren't very useful, looks like maybe a problem with vectors and Genymotion... Maybe stock emulators will work better. I'll try it out before merging.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java [hunk 1]
```diff
@@ -233,17 +232,14 @@ public void onClick(View view) {
...
                 @Override
                 public void onClick(View view) {
                     launchFeaturedMediaPicker();
                 }
             });
         } else {
-            featuredImageLabel.setVisibility(View.GONE);
-            mFeaturedImageView.setVisibility(View.GONE);
-            mFeaturedImageButton.setVisibility(View.GONE);
+            featuredImageCardView.setVisibility(View.GONE);
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: So much more convenient 😹

File: WordPress/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -459,4 +459,58 @@
 
     <!-- Plans Styles -->
     <style name="PlansProgressBar" parent="@android:style/Widget.Holo.ProgressBar" />
+
+    <!--Post Settings Styles-->
```
Comment (tonyr59h) [inline WordPress/src/main/res/values/styles.xml]: Nice job with the styling 👍

**Comment by tonyr59h** [comment]
> I like the idea of using explicit strings to make intent more clear so 👍 there. I did notice a bunch of duplicate strings that I think we should eliminate to reduce translation work. This isn't directed at this PR, just something I noticed while reviewing. Going to make an issue for it

#6066 created for this one.

**Comment by oguzkocer** [comment]
@tonyr59h Thanks a lot for the detailed tests. I've created a checklist for the issues you have reported [here](https://github.com/wordpress-mobile/WordPress-Android/issues/5811#issuecomment-308701774). I don't think they are in the scope of this PR. We could, of course, handle them here, but I think it's best to keep the PR to the point and not try to polish everything at once. I probably wasn't clear that this is still a WIP PR and the page is not yet polished. Anyhow, I think we should just merge this PR as-is, unless there is anything wrong with the changes (code-wise).

> Should the Status card have a header?

We removed it with @mattmiklic since we have a "Status" property for the post and it looked kind of weird having a Status label under the Status section.

**Comment by tonyr59h** [comment]
Thanks for making the checklist :shipit:

**Pull request merged**